From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: fukiame <fukiame@proton.me>
Date: Tue, 28 Jul 2026 05:29:01 +0700
Subject: [PATCH] lib: nav: tidy up single tab case

TODO fix empty thin slice when current song is null

Signed-off-by: fukiame <fukiame@proton.me>

diff --git a/lib/widgets/navigation/flick_nav_bar.dart b/lib/widgets/navigation/flick_nav_bar.dart
index ee93e6b62..b12156909 100644
--- a/lib/widgets/navigation/flick_nav_bar.dart
+++ b/lib/widgets/navigation/flick_nav_bar.dart
@@ -40,6 +40,9 @@ class _FlickNavBarState extends State<FlickNavBar> {
 
     final borderRadius = BorderRadius.circular(context.scaleSize(20));
 
+    bool sw = widget.showMiniPlayer && widget.miniPlayerWidget != null;
+    bool bl = buttons.length > 1;
+
     return AnimatedContainer(
       duration: AppConstants.animationNormal,
       curve: Curves.easeOutCubic,
@@ -49,6 +52,7 @@ class _FlickNavBarState extends State<FlickNavBar> {
         horizontalPadding,
         bottomPadding + verticalPadding,
       ),
+      padding: EdgeInsets.only(bottom: (sw && !bl) ? 4 : 0),
       decoration: BoxDecoration(
         gradient: LinearGradient(
           begin: Alignment.topLeft,
@@ -98,7 +102,8 @@ class _FlickNavBarState extends State<FlickNavBar> {
       child: Column(
         mainAxisSize: MainAxisSize.min,
         children: [
-          if (widget.showMiniPlayer && widget.miniPlayerWidget != null) widget.miniPlayerWidget!,
+          if (sw) widget.miniPlayerWidget!,
+          if (bl || !sw)
           AnimatedCrossFade(
             firstChild: _buildNavigationRow(context, buttons),
             secondChild: const SizedBox.shrink(),
-- 
2.55.0

